config/config.go

```diff
@@ -205,9 +205,9 @@ func (c *Config) Variables() map[string]interface{} {
 // This does not update the corresponding value in the Config struct.
 func (c *Config) Set(key string, val interface{}) {
 	c.m[key] = val
-	for _, item := range c.ms {
-		if item.Key == key {
-			item.Value = val
+	for i := range c.ms {
```

Comment:
1. Add a comment explaining why this is changed to repeatedly deference c.ms
2. Is there a guarantee that SetKey is properly used in thread-safe contexts?